### PR TITLE
SelfUpdateCommand - more details on download/copy failure.

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: src/Console/Command/SelfUpdateCommand.php [hunk 1]
```diff
@@ -109,14 +110,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $tempFilename = basename($localFilename, '.phar').'-tmp.phar';
         $remoteFilename = sprintf('https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/download/%s/php-cs-fixer.phar', $remoteTag);
 
+        $details = null;
         try {
-            $copyResult = @copy($remoteFilename, $tempFilename);
-            if (false === $copyResult) {
-                $output->writeln(sprintf('<error>Unable to download new version %s from the server.</error>', $remoteTag));
+            $copyResult = copy($remoteFilename, $tempFilename);
```
Comment (keradus) [inline src/Console/Command/SelfUpdateCommand.php]: `@` was silenting the warnings you got when sth goes wrong that couldn't be catched

File: src/Console/Command/SelfUpdateCommand.php [hunk 2]
```diff
@@ -109,14 +110,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $tempFilename = basename($localFilename, '.phar').'-tmp.phar';
         $remoteFilename = sprintf('https://github.com/FriendsOfPHP/PHP-CS-Fixer/releases/download/%s/php-cs-fixer.phar', $remoteTag);
 
+        $details = null;
         try {
```
Comment (julienfalque) [inline src/Console/Command/SelfUpdateCommand.php]: I think this `try/catch` is now useless as `copy()` does not throw exceptions.

Comment (keradus) [inline src/Console/Command/SelfUpdateCommand.php]: would be nice to double check it, maybe it does on 7.x

Comment (julienfalque) [inline src/Console/Command/SelfUpdateCommand.php]: It only triggers warnings and returns `false` (see https://3v4l.org/SatCS).

Comment (SpacePossum) [inline src/Console/Command/SelfUpdateCommand.php]: we don't transform warnings/errors to exceptions?

Comment (keradus) [inline src/Console/Command/SelfUpdateCommand.php]: as discussed, we don't.

**Comment by SpacePossum** [comment]
closing after discussion, cannot be done without BC break

**Pull request closed without merging**